### DISCUSSION TITLE
2FA - Don't ask anymore / Trusted device option

### DISCUSSION THREAD

**Discussion by bferrie93** [rustdesk/rustdesk#8513] [General]
I configured 2FA to connect to a client, it works well.
When I'm connected to this client, and when i want to transfer files, 2FA is asked again.
Is there a possibilty to not ask again 2FA to transfer files if there is already a connection to the client ?

Thanks

EDIT : I'll add: is it possible to add a "trusted device" option?

**Comment by rustdesk** [comment]
> EDIT : I'll add: is it possible to add a "trusted device" option?

Agree

**Reply by rustdesk** [reply]
@21pages add this option
- [x] use hwid https://github.com/tilda/rust-hwid for desktop, if failed use https://github.com/rustdesk/rustdesk/blob/eafebdba210cad50c837fbf6607df6e22e494ee8/libs/hbb_common/src/lib.rs#L295
- [x] valid in 90 days, remember (time, hwid || machine_id) on controlled side
- [x] clear if password changed
- [ ] how about if not enable`remember password`, but enable `trust this device`?

**Comment by rustdesk** [comment]
https://github.com/rustdesk/rustdesk/pull/9012/
